guides/cpp-intro.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: guide
+---
+
+# C++ introduction for the mbed developer
+
+The target audience for this guide is people who have not used C++ before and it will only cover the parts of C++ that is relevant for simple mbed programming.
+Understanding of programming languages in general is expected.
+
+# This code example
+
+This guide will use the following code example to explain how to use C++ with mbed.
+
+```cpp
+#include "mbed.h"
+
+DigitalOut myled(LED1);
+
+int main() {
+    while(1) {
+        myled = 1;
+        wait(0.4);
+        myled = 0;
+        wait(0.4);
+    }
+}
+```
+
+# Where general code is written
+
+When writing your code you write it inside the `int main()` function.
+This is the starting point for your program and every program will execute this function first.
+In our example the program will start by running the `while` loop.
+
+# Objects
+
+Objects in C++ works mostly like objects in other languages like Java.
+
+## Initialize object
+To intitalize an object you declare the object.
+
+```cpp
+DigitalOut myled(LED1);
+```
+
+This line creates a new object of type `DigitalOut` named `myled`.
+Writing that line will call the constructor of the object.
+In our case we are sending the constructor the parameter `LED1`.
+After this line we have the variable `myled` initialized with the parameter `LED1`.
+
+## Accessing attributes in objects
+
+To access an attribute on an object you use the dot notation.
+
+```cpp
+// access myAttribute of myObject
+myObject.myAttribute;
+
+int amount = myObject.count;
+
+// amount will now have the value of myObject.count
+printf("%d", amount);
+```
+
+## Calling functions on objects
+
+There is two ways of calling a function.
+If your object is a pointer you can call a function on an object use the arrow notation `->`.
```

Comment:
should be something like 'If your object is a pointer, you can call a function on the object by using the arrow notation'